lib/workos/sso.rb

```diff
@@ -168,6 +188,20 @@ def validate_domain_and_provider(domain:, provider:)
           " `provider` must be in #{PROVIDERS}"
       end
 
+      sig { params(response: Net::HTTPResponse).void }
+      def check_and_raise_error(response:)
```

Comment:
Noticed the client doesn't handle a 404. Would it helpful for this logic to be moved down there?

Reply:
Do you mean to where the request is executed so it covers all requests?

Reply:
Yep. Can't really think of any other cases where we might get a 404 but the client seems like a good home for that logic. Would getting a user potentially result in a 404 too if bad values are passed in?

Reply:
👍 good idea, I'll move it there

We'll return a 404 if the source doesn't exist or there's an authorization failure